sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AccessTokenManager.java

```diff
@@ -0,0 +1,230 @@
+     * Microsoft Graph API.
+     * @param permissions The expected permissions of resources.
+     * @return AccessToken
+     * @throws ServiceUnavailableException If fail to acquire the token.
+     */
+    public AccessToken getAccessToken(String applicationIdUri,
+                                      Set<String> permissions) throws ServiceUnavailableException {
+        AccessToken accessToken = loadAccessTokenFromSession(applicationIdUri);
+        Set<String> uniformedPermissions = permissions.stream()
+                                                      .map(String::trim)
+                                                      .map(s -> s.toLowerCase(Locale.ENGLISH))
+                                                      .filter(s -> !s.isEmpty())
+                                                      .collect(Collectors.toSet());
+        if (accessToken.permissions.containsAll(uniformedPermissions)) {
+            return accessToken;
+        } else {
+            // TODO: incremental consent.
+            String idToken = getIdTokenFromSecurityContext();
+            return getAccessToken(idToken, applicationIdUri, uniformedPermissions);
+        }
+    }
+
+    private String getIdTokenFromSecurityContext() {
+        return Optional.of(SecurityContextHolder.getContext())
+                       .map(SecurityContext::getAuthentication)
+                       .map(Authentication::getPrincipal)
+                       .map(p -> (OidcUser) p)
+                       .map(OidcUser::getIdToken)
+                       .map(AbstractOAuth2Token::getTokenValue)
+                       .orElse(null);
+    }
+
+    /**
+     * Acquire access token for a web-hosted resource. Only work for session supported web service.
+     *
+     * @param idToken The token used to perform an get token request.
+     * @param applicationIdUri The Application ID URI of web-hosted resource, e.g., https://graph.microsoft.com for
+     * Microsoft Graph API.
+     * @param permissions The permissions of resources to be authorized with, need to be formatted as lowercase.
+     * @return The access token for Graph service.
+     * @throws ServiceUnavailableException If fail to acquire the token.
+     * @throws MsalServiceException If {@link MsalServiceException} has occurred.
+     */
+    public AccessToken getAccessToken(String idToken,
+                                      String applicationIdUri,
+                                      Set<String> permissions) throws ServiceUnavailableException {
+        IAuthenticationResult result = getIAuthenticationResult(idToken, applicationIdUri, permissions);
+        Set<String> uniformedPermissions =
+            Arrays.stream(result.scopes().toLowerCase(Locale.ENGLISH).split(" "))
+                  .map(s -> s.startsWith(applicationIdUri) ? s.split(applicationIdUri)[1] : s)
+                  .collect(Collectors.toSet());
```

Comment:
Should the logic of save token into session be added here or somewhere else after token acquired? Currently this logic is called after token cache being refreshed only.